### PR TITLE
Improve unittest.mock.patch() types

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
pytype most likely fails for the same reason it failed for the Big Reformat. Should be fixed with the new pytype release. See also google/pytype#608.

File: stdlib/3/unittest/mock.pyi [hunk 1]
```diff
@@ -140,32 +140,60 @@ class CallableMixin(Base):
...
-    def __init__(self, getter: Any, attribute: Any, new: Any, spec: Any, create: Any, spec_set: Any, autospec: Any, new_callable: Any, kwargs: Any) -> None: ...
-    def copy(self) -> Any: ...
-    def __call__(self, func: Any) -> Any: ...
-    def decorate_class(self, klass: Any) -> Any: ...
-    def decorate_callable(self, func: Any) -> Any: ...
-    def get_original(self) -> Any: ...
+    @overload
+    def __init__(
+        self: _patch[MagicMock],
+        getter,
```
Comment (hauntsaninja) [inline stdlib/3/unittest/mock.pyi:159]: getter seems to be `Callable[[], Any]` and attribute is `str`

**Comment by srittau** [comment]
@hauntsaninja Undeferred and merged in master, let's see what CI says.

**Comment by srittau** [comment]
For some reason, travis and GitHub are not playing along. Travis CI is still running.

**Comment by srittau** [comment]
Travis CI passes.

**Comment by hauntsaninja** [comment]
Thanks! Willing to add `getter` as `Callable[[], Any]` and `attribute` as `str` to this?

**Comment by srittau** [comment]
@hauntsaninja I overlooked that comment. Added now.

**Pull request merged**